tokio/src/runtime/scheduler/multi_thread/queue.rs

```diff
@@ -394,7 +394,7 @@ impl<T> Steal<T> {
     /// Returns the number of entries in the queue
     pub(crate) fn len(&self) -> usize {
         let (_, head) = unpack(self.0.head.load(Acquire));
-        let tail = self.0.tail.load(Acquire);
+        let tail = self.0.tail.load(Relaxed);
```

Comment:
```suggestion
        let tail = self.0.tail.load(Acquire);
        let (_, head) = unpack(self.0.head.load(Relaxed));
```

I think the modification of `head` is the visible side-effect of `Release` modification of `tail`.

![queue](https://github.com/user-attachments/assets/87dcfe19-ac3a-465c-aa84-ece8251400e1)